samples/switchless/host/host.c

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/host.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "switchless_u.h"
+
+void host_increment(int* n)
+{
+    *n = *n + 1;
+}
+
+static bool check_simulate_opt(int* argc, const char* argv[])
+{
+    for (int i = 0; i < *argc; i++)
+    {
+        if (strcmp(argv[i], "--simulate") == 0)
+        {
+            fprintf(stderr, "Running in simulation mode\n");
+            memmove(&argv[i], &argv[i + 1], (*argc - i) * sizeof(char*));
+            (*argc)--;
+            return true;
+        }
+    }
+    return false;
+}
+
+int main(int argc, const char* argv[])
+{
+    oe_enclave_t* enclave = NULL;
+    oe_result_t result;
+    int ret = 1, m = 10000, n = 10000;
+    int oldm = m;
+
+    if (argc != 2 && argc != 3)
+    {
+        fprintf(stderr, "Usage: %s ENCLAVE_PATH [--simulate]\n", argv[0]);
+        return 1;
+    }
+
+    uint32_t flags = OE_ENCLAVE_FLAG_DEBUG;
+    if (check_simulate_opt(&argc, argv))
+    {
+        flags |= OE_ENCLAVE_FLAG_SIMULATE;
+    }
+
+    // Enable switchless and configure host worker number
+    oe_enclave_config_context_switchless_t config = {1, 0};
+    oe_enclave_config_t configs[] = {{
+        .config_type = OE_ENCLAVE_CONFIG_CONTEXT_SWITCHLESS,
+        .u.context_switchless_config = &config,
+    }};
+
+    if ((result = oe_create_switchless_enclave(
+             argv[1],
+             OE_ENCLAVE_TYPE_SGX,
+             flags,
+             configs,
+             OE_COUNTOF(configs),
+             &enclave)) != OE_OK)
+        fprintf(stderr, "oe_create_enclave(): result=%u", result);
+
+    // Call into the enclave
+    result = enclave_add_N(enclave, &m, n);
+    if (result != OE_OK)
```

Comment:
Here is some of my thoughts about this sample:
The readme has good explanation on how to make best of this switchless feature for certain scenarios (such on frequent calls and short host operations).
To give a user better idea on the benefit, I suggest have two separate ocalls: one with switchless and one without , make them call back to the host for same numbers of times (such as 100 or 1000) on the same simple task (like add_N you are doing), then print out the time take it and compare them. I am pretty it would give developers a better sense on this feature's benefits on top of how to use it.

Reply:
Sounds good to me.

Reply:
Good suggestion. In fact `tests/switchless` is implemented this way. I will port the code to this sample.